e2etests/test/device_acknowledge_tests.js

```diff
@@ -6,16 +6,50 @@
 var serviceSdk = require('azure-iothub');
 var createDeviceClient = require('./testUtils.js').createDeviceClient;
 var closeDeviceServiceClients = require('./testUtils.js').closeDeviceServiceClients;
+var DeviceIdentityHelper = require('./device_identity_helper.js');
 
 var assert = require('chai').assert;
 var debug = require('debug')('e2etests');
 var uuid = require('uuid');
 
-var runTests = function (hubConnectionString, deviceTransport, provisionedDevice) {
-  describe('Device utilizing ' + provisionedDevice.authenticationDescription + ' authentication, connected over ' + deviceTransport.name, function () {
+var deviceAmqp = require('azure-iot-device-amqp');
+var deviceHttp = require('azure-iot-device-http');
+
+var hubConnectionString = process.env.IOTHUB_CONNECTION_STRING;
+
+[
+  DeviceIdentityHelper.createDeviceWithSas,
+  DeviceIdentityHelper.createDeviceWithSymmetricKey,
+  DeviceIdentityHelper.createDeviceWithX509SelfSignedCert
+].forEach(function (createDeviceMethod) {
+  [
+    deviceHttp.Http,
+    deviceAmqp.Amqp,
+    deviceAmqp.AmqpWs
+  ].forEach(function (deviceTransport) {
```

Comment:
>forEach [](start = 4, length = 7)

Creating the device before calling describe could slow the tests down because it forces us to pass --delay to mocha.  I don't know if the delay is significant or not